### ISSUE TITLE
Lit Tests: Large binaries and long build times

### ISSUE THREAD

**Issue by e3m3** [e3m3/mlir-rust#1]
Currently, the lit tests sub-projects are linked against the main library crate statically, causing lots of duplicated code as each rust lit test is built as an isolated binary.
* This causes rebuilds of the lit tests `cargo test -- --nocapture` to take a long time (2min+).
* The target directory can be quite large (20GB+) given the number of lit tests (200+).
* The test build for the sub-project must currently be triggered indirectly within the rust test hook as a call to a cargo build sub-process. Otherwise, the user must build the test sub-project separately before calling `cargo test` at the top-level.

Using cargo workspaces has slightly alleviated the problem by merging the target directory for both the library and sub-project test binaries.

**Comment by e3m3** [comment]
On an Apple M2:
```
********************                         

Testing Time: 208.90s                        

Total Discovered Tests: 257                  
  Passed           : 253 (98.44%)            
  Expectedly Failed:   4 (1.56%)             

test tests::lit ... ok                       

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 281.52s
```

**Comment by e3m3** [comment]
Fixed since Commit f6ee170.

Closing.
